### PR TITLE
Feature/sub navbar

### PR DESCRIPTION
**Pull request by geemanjs**
#4

### CODE REVIEW

**Comment by lennym** [comment]
The example page throws an error at the moment:

```
TypeError: element.type.toLowerCase is not a function
    at ReactDOMServerRenderer.renderDOM (/Users/lenny/dev/govuk-react-components/example/node_modules/react-dom/cjs/react-dom-server.node.development.js:2304:28)
    at ReactDOMServerRenderer.render (/Users/lenny/dev/govuk-react-components/example/node_modules/react-dom/cjs/react-dom-server.node.development.js:2298:21)
    at ReactDOMServerRenderer.read (/Users/lenny/dev/govuk-react-components/example/node_modules/react-dom/cjs/react-dom-server.node.development.js:2234:19)
    at Object.renderToStaticMarkup (/Users/lenny/dev/govuk-react-components/example/node_modules/react-dom/cjs/react-dom-server.node.development.js:2512:25)
    at View.renderFile [as engine] (/Users/lenny/dev/govuk-react-components/example/node_modules/express-react-views/index.js:60:32)
    at View.render (/Users/lenny/dev/govuk-react-components/example/node_modules/express/lib/view.js:135:8)
    at tryRender (/Users/lenny/dev/govuk-react-components/example/node_modules/express/lib/application.js:640:10)
    at Function.render (/Users/lenny/dev/govuk-react-components/example/node_modules/express/lib/application.js:592:3)
    at ServerResponse.render (/Users/lenny/dev/govuk-react-components/example/node_modules/express/lib/response.js:1008:7)
    at app.get (/Users/lenny/dev/govuk-react-components/example/index.js:19:7)
```

File: src/subnavbar/navitem/index.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    className: PropTypes.string,
+    active: PropTypes.bool
+  };
+  return component;
+};
+
+export const NavItem = navItemOf('a');
+NavItem.propTypes = Object.assign({ href: PropTypes.string.isRequired }, NavItem.propTypes);
+
+export default NavItem;
```
Comment (lennym) [inline src/subnavbar/navitem/index.jsx:26]: This code feels a bit convoluted when only one type is ever used. Could we refactor to get rid of that type abstraction, or is there a plan for it?

**Comment by lennym** [comment]
Is there a reason the test code isn't inside a `__tests__` directory as per other components?

Edit: "as per _most_ other components" - we might want to make this consistent one way or the other.

File: src/subnavbar/navitem/index.jsx (hunk 1 above)
Comment (geemanjs) [reply src/subnavbar/navitem/index.jsx:26]: The type abstraction is pretty useful if you want to use something like `'react-router'` with this (my use case for this library)

You define a different `NavItem` styled in the NavItem way:
```jsx
import React from 'react';
import {Link} from 'react-router';
import {navItemOf} from 'govuk-react-components/'

export default navItemOf(Link);
```

Comment (lennym) [reply src/subnavbar/navitem/index.jsx:26]: I see. That's fine then.

File: example/views/index.jsx [hunk 2]
```diff
@@ -1,10 +1,11 @@
 const React = require('react');
 const Layout = require('./layout');
-
 const Input = require('../../components/forms/input-text');
 const RadioGroup = require('../../components/forms/radio-group');
 const Select = require('../../components/forms/select');
 const Date = require('../../components/forms/date');
+const SubNavBar = require('../../components/subnavbar/index');
+const NavItem = require('../../components/subnavbar/navitem').NavItem;
```
Comment (lennym) [inline example/views/index.jsx]: Changing these two lines as follows fixes the error:

```js
const {SubNavBar} = require('../../components/subnavbar');
const {NavItem} = require('../../components/subnavbar/navitem');
```

**Comment by geemanjs** [comment]
@lennym all should be fixed now. Forgot how strict node is for import/export

File: example/views/index.jsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 const React = require('react');
 const Layout = require('./layout');
-
 const Input = require('../../components/forms/input-text');
 const RadioGroup = require('../../components/forms/radio-group');
 const Select = require('../../components/forms/select');
 const Date = require('../../components/forms/date');
+const SubNavBar = require('../../components/subnavbar').default;
```
Comment (lennym) [inline example/views/index.jsx:7]: It's _really_ nasty IMO that this needs a `.default` :-(

Comment (geemanjs) [reply example/views/index.jsx:7]: I agree.. Is there a better way to expose multiple exports?

Basically want the 'default' to be the component and then additional items available as well.

Below is how this looks on a 'create-react-app' project. 

```jsx
import SubNavBar from '../../components/subnavbar';

// or .. 
import { SubNavBar, NavItem } from '../../components/subnavbar
```

Comment (lennym) [reply example/views/index.jsx:7]: What I meant to say before I left the office and got distracted, was that I think it's possible to use ES modules in the views, and the react view engine will compile them on the fly. That way jsx just always uses `import` and it "just works".

I'll give it a try.

Comment (geemanjs) [reply example/views/index.jsx:7]: ^^ OK awesome.. That'd be good!

**Pull request merged**